### PR TITLE
add module

### PR DESCRIPTION
**Pull request by guojian0614**
add some module

### CODE REVIEW

File: sparrow-blog-dao/pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        <maven.compiler.target>8</maven.compiler.target>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>com.sparrow.blog</groupId>
+            <artifactId>sparrow-blog-protocol</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
```
Comment (sparrowzoo) [inline sparrow-blog-dao/pom.xml]: dao 是接口定义 除protocol 外不应该依赖其他包

Comment (guojian0614) [reply sparrow-blog-dao/pom.xml]: correct already finished

**Comment by guojian0614** [comment]
corrent the depend suggest

**Comment by guojian0614** [comment]
add base config

File: sparrow-blog-dao-impl/src/main/java/com/sparrow/blog/dao/sharding/strategy/LongHashSharding.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    private static long getIndex(long size, Long value) {
+        final long hash = hash(value);
+        final long i = hash & (size - 1L);
+        return i;
+    }
+
+    private static final long hash(long hashCode) {
+        long h;
+        return  (h = hashCode) ^ (h >>> 16);
```
Comment (bigduu) [inline sparrow-blog-dao-impl/src/main/java/com/sparrow/blog/dao/sharding/strategy/LongHashSharding.java:30]: Is it necessary to define `h`?

File: sparrow-blog-dao/src/main/java/com/sparrow/blog/dao/dos/PostsEntityDO.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+import com.baomidou.mybatisplus.annotation.IdType;
+import com.baomidou.mybatisplus.annotation.TableField;
+import com.baomidou.mybatisplus.annotation.TableId;
+import com.baomidou.mybatisplus.annotation.TableName;
+
+import java.util.Date;
+
+@TableName("posts")
+public class PostsEntityDO {
```
Comment (bigduu) [inline sparrow-blog-dao/src/main/java/com/sparrow/blog/dao/dos/PostsEntityDO.java:12]: I think there is a missing `@Data` annotation
```suggestion
@Data
@TableName("posts")
public class PostsEntityDO {
```

File: sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                return parameter.getParameter().getParameterizedType().equals(ClientInformation.class);
+            }
+
+            @Override public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer container,
+                NativeWebRequest request, WebDataBinderFactory factory) throws Exception {
+                HttpServletRequest req = (HttpServletRequest) request.getNativeRequest();
+                ClientInformation client = new ClientInformation();
+                //todo 解析 client 自定义参数
+                client.setIp("127.0.0.1");
+                return client;
```
Comment (bigduu) [inline sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java:42]: I don't know why you get the `NativeRequets` but do not use or return them.

File: sparrow-blog-main/src/main/java/com/sparrow/blog/controller/PostsController.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import com.sparrow.blog.service.PostsSevice;
+import com.sparrow.protocol.Result;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("posts")
+public class PostsController extends BaseController {
+    @Autowired
+    private PostsSevice postsSevice;
```
Comment (bigduu) [inline sparrow-blog-main/src/main/java/com/sparrow/blog/controller/PostsController.java:16]: Please use the constructor to inject the `bean`
```suggestion
     private PostsSevice postsSevice;
    public PostsController(PostsSevice postsSevice) {
        this.postsSevice = postsService;
    }
```

File: sparrow-blog-main/src/main/resources/application-local.properties [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#
+spring.shardingsphere.datasource.master.url=jdbc:mysql://localhost:3306/sparrow_blog?useUnicode=true&characterEncoding=utf8&autoReconnect=true&allowMultiQueries=true&useSSL=false&serverTimezone=GMT%2B8
+spring.shardingsphere.datasource.master.username=root
+spring.shardingsphere.datasource.master.password=root
```
Comment (bigduu) [inline sparrow-blog-main/src/main/resources/application-local.properties:4]: Need to set the password to environment value

**Review by bigduu — COMMENTED** [review]
There are some comments added, I've not much knowledge for `mybatis` please review it 
@sparrowzoo

File: sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java (hunk 4 above)
Comment (bigduu) [reply sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java:42]: And could you please extract the `127.0.0.1` to the static value or move it into properties?

File: sparrow-blog-main/src/main/resources/application-local.properties (hunk 6 above)
Comment (guojian0614) [reply sparrow-blog-main/src/main/resources/application-local.properties:4]: > 

this is local config

File: sparrow-blog-main/src/main/java/com/sparrow/blog/controller/PostsController.java (hunk 5 above)
Comment (guojian0614) [reply sparrow-blog-main/src/main/java/com/sparrow/blog/controller/PostsController.java:16]: officely just recommend it ，not mandatory。

File: sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java (hunk 4 above)
Comment (guojian0614) [reply sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java:42]: this is copy from password ddd project，i don‘t know the meaning of this code.

File: sparrow-blog-dao/src/main/java/com/sparrow/blog/dao/dos/PostsEntityDO.java (hunk 3 above)
Comment (guojian0614) [reply sparrow-blog-dao/src/main/java/com/sparrow/blog/dao/dos/PostsEntityDO.java:12]: I want not write getter setter method per field everyone.

File: sparrow-blog-dao-impl/src/main/java/com/sparrow/blog/dao/sharding/strategy/LongHashSharding.java (hunk 2 above)
Comment (guojian0614) [reply sparrow-blog-dao-impl/src/main/java/com/sparrow/blog/dao/sharding/strategy/LongHashSharding.java:30]: just copy from hashmap ,try to keep it as it is.

File: sparrow-blog-dao/src/main/java/com/sparrow/blog/dao/dos/PostsEntityDO.java (hunk 3 above)
Comment (bigduu) [reply sparrow-blog-dao/src/main/java/com/sparrow/blog/dao/dos/PostsEntityDO.java:12]: That's cool, how do others use the Class? I see all fields are private permission to access.

File: sparrow-blog-dao-impl/src/main/java/com/sparrow/blog/dao/sharding/strategy/LongHashSharding.java (hunk 2 above)
Comment (bigduu) [reply sparrow-blog-dao-impl/src/main/java/com/sparrow/blog/dao/sharding/strategy/LongHashSharding.java:30]: A small question, could we use a general method to calculate the hash value?

File: sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java (hunk 4 above)
Comment (bigduu) [reply sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java:42]: @sparrowzoo There is an unknown code, could you please help to explain why the request is not used and not rerun?

File: sparrow-blog-main/src/main/java/com/sparrow/blog/controller/PostsController.java (hunk 5 above)
Comment (bigduu) [reply sparrow-blog-main/src/main/java/com/sparrow/blog/controller/PostsController.java:16]: Yeah, I know it, but I still recommended remove the project warning code

**Pull request merged**